### ISSUE TITLE
2b2t anti afk not working

### ISSUE THREAD

**Issue by RegenixMC** [themoonisacheese/2bored2wait#192]
It would be cool if you guys could make the 2b2t anti afk work.

**Comment by MrGeorgen** [comment]
yeah we could let the player walk around by using mineflayer to connect to 2b2w

**Comment by MrGeorgen** [comment]
it is not fixed


> yeah we could let the player walk around by using mineflayer to connect to 2b2w. we could resue some code from here https://gist.github.com/rvdm/0c4b0be1d3d49313c9c9fc69b5d9bad7

that was only an idea

**Comment by preludium975** [comment]
I'd appreciate an anti-AFK feature too.

**Comment by cristiioan** [comment]
> yeah we could let the player walk around by using mineflayer to connect to 2b2w. we could resue some code from here https://gist.github.com/rvdm/0c4b0be1d3d49313c9c9fc69b5d9bad7
@MrGeorgen I have something like this in devolopment on a fork of 2bored2wait(remake edition for more flexibility)

**Comment by MrGeorgen** [comment]
> @MrGeorgen I have something like this in devolopment on a fork of 2b2t_rewrite

I can't find your fork

**Comment by cristiioan** [comment]
Is private.

**Comment by agi0rgi** [comment]
A lovely feature would be to have the bot to start fishing as soon as it joins the server, this way you should not be kicked for being AFK

**Comment by plebeo** [comment]
Yes, please. This would be the most appreciated feature for a lot of users.

**Comment by Doogie13** [comment]
Try using kami blue as a reference. Its antiafk works.

**Comment by AverageComet250** [comment]
Is this still planned, or is it in the works? If it isn't I might be able to take this up and create a PR.

**Comment by MrGeorgen** [comment]
> Is this still planned, or is it in the works? If it isn't I might be able to take this up and create a PR.

I planned doing it at some point, but I'm not currently working on it.
If you want to make a pr for it, please note that mineflayer was not designed
for a proxy. This can cause issues in the phyics engine, but I don't know
anymore where I have this information from. Anyway this means you should not
use the mineflayer client that is connected to 2b2t. Another mineflayer client
should connected to the 2b2w proxy. When using the gist obove as a refence, it
should be fairly straigh forward. I also can't say for sure this anti-afk
works on 2b2t

**Comment by AverageComet250** [comment]
I've took a look at the anti afk in Kami blue and it's seems to work pretty well (I set up a server with a 5 mins time out and I stayed connected for a few hours). It seems to just use baritone to move the player, jump and swing the arm every so often. That should be pretty easy to implement

Thanks,
Shravan Mandava

________________________________

**Comment by czho** [comment]
This probably won't work, if you didnt know baritone cant run on mineflayer there completely incompatible and the replacement to baritone that works on mineflayer dosent work with the 2b2t anticheat.

**Comment by AverageComet250** [comment]
I know I can't run baritone on mineflayer, but I can create something similar with the bare minimum for what we need.

I just have to make a piece of code that walks, turns, jumps and swings a user's arm every so often.

**Comment by MrGeorgen** [comment]
> I know I can't run baritone on mineflayer, but I can create something similar with the bare minimum for what we need.
> 
> I just have to make a piece of code that walks, turns, jumps and swings a user's arm every so often.
> 
> And that can all run at the mineflayer client level since the user will be disconnected.

This is exactlly what the gist does  https://gist.github.com/rvdm/0c4b0be1d3d49313c9c9fc69b5d9bad7

**Comment by AverageComet250** [comment]
In that case, it's as simple as replacing the current anti-afk with a modified version of this script.

I'd do it myself and create a pr, but I'm currently working on re-doing the website with react and adding more features to it.

**Comment by etiaro** [comment]
Hi, I am planning to start playing on 2b2t and found this repo while I was writing something similar myself. It's really nice project, i'm interested helping You. Is anyone working on this antiafk? If not, I can try to write it down

**Comment by AverageComet250** [comment]
Yeah sure go for it. I was gonna implement it myself but I'm currently working on upgrading the web page

Thanks,
Shravan Mandava

________________________________

**Comment by etiaro** [comment]
#273 walking antiAFK is done, but needs some testing on 2b2t(I just took a look if works on localhost).

**Comment by rob9315** [comment]
@RegenixMC cool to see you here :laughing:
@etiaro 2b2t uses a little more advanced anticheat
@MrGeorgen real anti-afk is possible, since now there is a mineflayer bot running at all time, you will only have to kick the client and then make the bot do stuff to circumvent the antiafk, though as I said, 2b2t's anticheat is a little more than just jumping and walking in a circle.

**Comment by etiaro** [comment]
@rob9315 any suggestions? For now there's random timing of randomly picked yaw&pitch and/or jump and/or walking direction, divided by random intervals of staying and doing nothing. I am now testing if adding right-click to these actions could increase afk time, and also checking if fishing can do the trick(when i'll get myself a fishing rod with mending).  I'll let you know about progress when I'll get past the queue again

**Comment by AverageComet250** [comment]
I think we'll need someone with prio at this rate.

Make the time between actions a config option @etiaro and set it to the same as Kami blur's default

**Comment by etiaro** [comment]
I would rewrite code into async/await to make code clear, add placing/destroying blocks, swimming in water, everything configurable, but i'm a little out of time next week. Current version with walking/clicking/fishing is available at https://github.com/etiaro/2bored2wait/tree/fishing

**Comment by etiaro** [comment]
@AverageComet250 I see Kami blue is swinging arm/jumping/randomly turning and walking to square corners with baritone.

**Comment by HyCore** [comment]
antiafk still doesn't work on last release
